packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -2538,7 +2560,23 @@ function commitMutationEffectsDeletions(
         captureCommitPhaseError(childToDelete, error);
       }
     }
-    // Don't clear the Deletion effect yet; we also use it to know when we need to detach refs later.
+
+    // If there are no pending passive effects, it's safe to detach remaining pointers now.
+    const primarySubtreeTag = childToDelete.subtreeTag & PassiveSubtreeTag;
+    const primaryEffectTag = childToDelete.effectTag & PassiveMask;
+    if (primarySubtreeTag === NoSubtreeTag && primaryEffectTag === NoEffect) {
+      detachFiberAfterEffects(childToDelete);
```

Comment:
Assuming we always schedule a passive effect for deletions, don't need this block now.

Reply:
Yes. I know. Adding `Deletion` to the static mask means we can remove 4 of these places. ;)

Reply:
Ah, yeah fair enough. It is safe to remove now.

I'm a tiny bit concerned that it's not intuitive how re-adding that optimization would impact a few of these places, but maybe that won't actually be a concern.